src/motion/usePointCompare.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useMemo } from 'react';
+
+import type { ObservablePoint, PointData } from 'pixi.js';
+
+type NumberPoint = number | ObservablePoint | PointData | undefined;
+
+const toDependencyArray = (point: NumberPoint) => [
+    typeof point === 'object' ? point.x : null,
+    typeof point === 'object' ? point.y : null,
+    typeof point !== 'object' ? point : null,
+];
+
+export const usePointCompareEffect = (
+    callback: VoidFunction,
+    point: NumberPoint
+) =>
+    useEffect(callback, toDependencyArray(point));
```

Comment:
tbh, I had no idea that the dependency array itself did not have to be referentially stable, only it's contents. til.